lib/matplotlib/cbook.py

```diff
@@ -2192,6 +2192,21 @@ def is_math_text(s):
     return even_dollars
 
 
+def _check_1d(x):
+    '''
+    Converts a sequence of less than 1 dimension, to an array of 1
+    dimension; leaves everything else untouched.
+    '''
```

Comment:
What about instead using something like the following?
```python
def asarray_units_okay(x):
    if type(x) in units.registry:
        return x
    else:
        return np.asarray(x)

def atleast_1d_units_okay(x);
    x = asarray_units_okay(x)
    if x.ndim == 0:
        x = x[np.newaxis]
    return x
```

I am nervous about preemptively checking all allowed operations.